rllib/agents/dqn/simple_q.py

```diff
@@ -49,120 +50,223 @@
+        grad_clip: Optional[int] = None,
+        learning_starts: Optional[int] = None,
+        **kwargs,
+    ) -> "SimpleQConfig":
+        """Sets the training related configuration.
+        Args:
+            timesteps_per_iteration: Minimum env steps to optimize for per train call.
+            This value does not affect learning, only the length of iterations.
+            target_network_update_freq: Update the target network every
+                `target_network_update_freq` steps.
+            replay_buffer_config: Replay buffer config.
+            store_buffer_in_checkpoints: Set this to True, if you want the contents
+                of your buffer(s)
+            to be stored in any saved checkpoints as well.
+                Warnings will be created if:
+                    - This is True AND restoring from a checkpoint that contains no
+                      buffer data.
+                    - This is False AND restoring from a checkpoint that does contain
+                      buffer data.
+            lr_schedule: Learning rate schedule. In the format of
+                [[timestep, value], [timestep, value], ...].
+                A schedule should normally start from timestep 0.
+            adam_epsilon: Adam epsilon hyper parameter
+            grad_clip: If not None, clip gradients during optimization at this value
+            learning_starts: How many steps of the model to sample before learning
+                starts.
+
+        Returns:
+            This updated TrainerConfig object.
+        """
+        # Pass kwargs onto super's `training()` method.
+        super().training(**kwargs)
+
+        if timesteps_per_iteration is not None:
+            self.timesteps_per_iteration = timesteps_per_iteration
+        if target_network_update_freq is not None:
+            self.target_network_update_freq = target_network_update_freq
+        if replay_buffer_config is not None:
+            self.replay_buffer_config = replay_buffer_config
+        if store_buffer_in_checkpoints is not None:
+            self.store_buffer_in_checkpoints = store_buffer_in_checkpoints
+        if lr_schedule is not None:
+            self.lr_schedule = lr_schedule
+        if adam_epsilon is not None:
+            self.adam_epsilon = adam_epsilon
+        if grad_clip is not None:
+            self.grad_clip = grad_clip
+        if learning_starts is not None:
+            self.learning_starts = learning_starts
+
+        return self
+
+
+# Deprecated: Use ray.rllib.agents.dqn.simple_q.SimpleQConfig instead!
+class _deprecated_default_config(dict):
+    def __init__(self):
+        super().__init__(SimpleQConfig().to_dict())
+
+    @Deprecated(
+        old="ray.rllib.agents.dqn.simple_q.DEFAULT_CONFIG",
+        new="ray.rllib.agents.dqn.simple_q.SimpleQConfig(...)",
+        error=False,
+    )
+    def __getitem__(self, item):
+        return super().__getitem__(item)
+
+
+DEFAULT_CONFIG = _deprecated_default_config()
```

Comment:
Same here, move all the way down to the bottom of the file, such that SimpleQTrainer is already defined when we construct the old-school config dict.